client/src/components/advancedSelectWidget/MultiTypeAdvancedSelectComponent.js

```diff
@@ -0,0 +1,170 @@
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Report.autocompleteQuery,
+  addon: REPORTS_ICON
+}
+
+const widgetPropsPeople = {
+  objectType: Models.Person,
+  overlayRenderRow: PersonDetailedOverlayRow,
+  overlayColumns: ["Name", "Position", "Location", "Organization"],
+  filterDefs: peopleFilters,
+  queryParams: {},
+  fields: Models.Person.autocompleteQueryWithNotes,
+  addon: PEOPLE_ICON
+}
+
+const widgetPropsOrganization = {
+  objectType: Models.Organization,
+  overlayRenderRow: OrganizationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Organization.autocompleteQuery,
+  addon: ORGANIZATIONS_ICON
+}
+
+const widgetPropsPosition = {
+  objectType: Models.Position,
+  overlayRenderRow: PositionOverlayRow,
+  overlayColumns: ["Position", "Organization", "Current Occupant"],
+  filterDefs: entityFilters,
+  queryParams: {},
+  fields: Models.Position.autocompleteQuery,
+  addon: POSITIONS_ICON
+}
+
+const widgetPropsLocation = {
+  objectType: Models.Location,
+  overlayRenderRow: LocationOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Location.STATUS.ACTIVE },
+  fields: Models.Location.autocompleteQuery,
+  addon: LOCATIONS_ICON
+}
+
+const widgetPropsTask = {
+  objectType: Models.Task,
+  overlayRenderRow: TaskSimpleOverlayRow,
+  overlayColumns: ["Name"],
+  filterDefs: entityFilters,
+  queryParams: { status: Models.Task.STATUS.ACTIVE },
+  fields: Models.Task.autocompleteQueryWithNotes,
+  addon: TASKS_ICON
+}
+
+const widgetTypeMapping = {
+  [ENTITY_TYPES.REPORTS]: widgetPropsReport,
+  [ENTITY_TYPES.PEOPLE]: widgetPropsPeople,
+  [ENTITY_TYPES.ORGANIZATIONS]: widgetPropsOrganization,
+  [ENTITY_TYPES.POSITIONS]: widgetPropsPosition,
+  [ENTITY_TYPES.LOCATIONS]: widgetPropsLocation,
+  [ENTITY_TYPES.TASKS]: widgetPropsTask
+}
+
+const MultiTypeAdvancedSelectComponent = ({
+  onConfirm,
+  objectType: objectTypeArg
+}) => {
+  const [objectType, setObjectType] = useState(
+    objectTypeArg || ENTITY_TYPES.REPORTS
+  )
+  const [advancedSelectProps, setAdvancedSelectProps] = useState(
+    widgetTypeMapping[objectType]
+  )
+
+  function changeObjectType(newObjectType) {
+    setObjectType(newObjectType)
+    setAdvancedSelectProps(widgetTypeMapping[newObjectType])
+  }
+
+  return (
+    <>
+      <ButtonToggleGroup value={objectType} onChange={changeObjectType}>
+        {Object.entries(ENTITY_TYPES).map((key, value) => {
+          const entityName = key[1]
+          const entityLabel = SEARCH_OBJECT_LABELS[key[0]]
+          return (
+            <Button key={entityName} value={entityName}>
+              {entityLabel}
+            </Button>
+          )
+        })}
+      </ButtonToggleGroup>
+
+      <AdvancedSingleSelect
+        autofocus="true"
+        fieldName="entitySelect"
+        fieldLabel="Search in ANET:"
+        placeholder={"Find " + objectType.toLowerCase()}
```

Comment:
Note: this will show _"Find task"_ for objectives/efforts.

Reply:
Fixed that and did some more refactoring of the MultiTypeAdvancedSelectComponent